### ISSUE TITLE
False positive on shape check in default when an object global var is used

### ISSUE THREAD

**Issue by elprans** [edgedb/edgedb#4369]
```
  global current_organization_id -> uuid;
  global current_organization := (
    select Organization filter .id = global current_organization_id
  );

  type Organization;

  type SomeOrgSpecificObject {
    required link organization -> Organization {
      default := global current_organization;
    }
  } 
```

```
error: SchemaDefinitionError: default expression may not include a shape
   ┌─ query:11:7
   │
11 │       default := global current_organization;
   │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error
```

**Comment by mkhlsvn** [comment]
A very unpleasant bug, the consequence of which is the redundancy of the model code.

**Comment by divan** [comment]
Is there any update on this bug?

As soon as I add globals to schema, the abovementioned example becomes very tempting and useful to use (set defaults from globals). But this bug prevents it from using completely.

**Comment by mkhlsvn** [comment]
For edgedb 3.0 the bug is still relevant. When can we expect a fix?

**Comment by msullivan** [comment]
Fix will go out in 3.4
